google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2922,6 +2962,40 @@ boolean isValid() {
     }
   }
 
+  /**
+   * Returns a multiplexed session. We would always return the session which is at the front of the
+   * queue since this will be the most recently created session.
+   */
+  SessionFuture getMultiplexedSessionWithFallback() throws SpannerException {
+    if (options.getUseMultiplexedSession()) {
+      ISpan span = tracer.getCurrentSpan();
+      span.addAnnotation("Acquiring multiplexed session");
+      synchronized (lock) {
+        try {
+          SessionFuture sessionFuture = getMultiplexedSession();
```

Comment:
This method should not be called while holding the lock, as it is potentially blocking (it calls `Future.get()`. It is also not necessary to hold the lock to call this method, as both `AtomicReference` and `SettableApiFuture` are thread safe.

Reply:
Removed